src/main/services/downloader.ts

```diff
@@ -234,6 +237,9 @@ export class Downloader {
                         outputStream.end(null, null, () => {
                             return resolve(download);
                         });
+                    } else {
+
+                        return reject("output stream error");
```

Comment:
This should be `resolve()`, not reject (because even though `outputStream` is non-existent, we are still producing a valid `download` data structure).

Reply:
There is a `reject()` missing in `if (response.statusCode < 200 || response.statusCode > 299) {` ... I am implementing several bug fixes, I will cherry-pick the `agentOptions` / `rejectUnauthorized` solution from your PR.